hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/BlockContentsValidator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.junit.support.validators.block;
+
+import static com.hedera.services.bdd.junit.hedera.utils.WorkingDirUtils.workingDirFor;
+
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.services.bdd.junit.support.BlockStreamAccess;
+import com.hedera.services.bdd.junit.support.BlockStreamValidator;
+import com.hedera.services.bdd.spec.HapiSpec;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Paths;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Assertions;
+
+public class BlockContentsValidator implements BlockStreamValidator {
+    private static final Logger logger = LogManager.getLogger(BlockContentsValidator.class);
+
+    public static void main(String[] args) {
+        final var node0Dir = Paths.get("hedera-node/test-clients")
+                .resolve(workingDirFor(0, "hapi"))
+                .toAbsolutePath()
+                .normalize();
+        final var validator = new BlockContentsValidator();
+        final var blocks =
+                BlockStreamAccess.BLOCK_STREAM_ACCESS.readBlocks(node0Dir.resolve("data/block-streams/block-0.0.3"));
+        validator.validateBlocks(blocks);
+    }
+
+    public static final Factory FACTORY = new Factory() {
+        @NonNull
+        @Override
+        public BlockStreamValidator create(@NonNull final HapiSpec spec) {
+            return new BlockContentsValidator();
+        }
+    };
+
+    @Override
+    public void validateBlocks(@NonNull final List<Block> blocks) {
+        for (int i = 0; i < blocks.size(); i++) {
+            try {
+                validate(blocks.get(i), i == 0);
+            } catch (AssertionError err) {
+                logger.error("Error validating block {}", blocks.get(i));
+                throw err;
+            }
+        }
+    }
+
+    private static void validate(Block block, boolean isFirst) {
```

Comment:
Is the argument `isFirst` necessary? Did you have an idea to use this in the future?

Reply:
Removed.